### PR TITLE
topologyinfo: add unit tests

### PR DESCRIPTION
**Pull request by ffromani**
Improve the test coverage
To actually test the sysfs helper package we introduce an
additional subpackage to fake sysfs tree - with its own tests.

Signed-off-by: Francesco Romani <fromani@redhat.com>

### CODE REVIEW

File: pkg/topologyinfo/numa/distances/distances.go [hunk 1]
```diff
@@ -65,22 +65,44 @@ func (d *Distances) BetweenNodes(from, to int) (int, error) {
 	return d.byNode[from].values[to], nil
```
Comment (swatisehgal) [inline pkg/topologyinfo/numa/distances/distances.go]: We can make byNode a 2D array. WDYT?

File: pkg/topologyinfo/sysfs/fake/fakesysfs.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+type FakeSysfs struct {
+	base string
+	root Tree
+}
+
+func NewFakeSysfs(base string) (*FakeSysfs, error) {
+	return &FakeSysfs{
+		base: base,
+		// DO NOT USE NEITHER "." or "" HERE!!
+		root: NewTree("_", nil),
```
Comment (swatisehgal) [inline pkg/topologyinfo/sysfs/fake/fakesysfs.go]: Initially the tree is meant to be empty so root should be &tree{} but that's not the case here, why is the name here "_" here?

**Review by swatisehgal — CHANGES_REQUESTED** [review]
Very nice PR! just a few questions/suggestions

File: pkg/topologyinfo/numa/distances/distances.go (hunk 1 above)
Comment (ffromani) [reply pkg/topologyinfo/numa/distances/distances.go]: Considering that the cost matrix is dense, this is probably a good idea.

File: pkg/topologyinfo/sysfs/fake/fakesysfs.go (hunk 2 above)
Comment (ffromani) [reply pkg/topologyinfo/sysfs/fake/fakesysfs.go]: nice catch, this seems a relic of initial implementation. We need at very least a clearer explanation should "_" be needed, so I'll just remove it now - for the reasons you said, and document/reintroduce later if needed.

**Comment by swatisehgal** [comment]
LGTM! Merging it now.

**Pull request merged**